src/platform_impl/android/mod.rs

```diff
@@ -551,25 +395,48 @@ impl<T: 'static> EventLoop<T> {
                     }
                 }
                 InputEvent::KeyEvent(key) => {
-                    let device_id = event::DeviceId(DeviceId);
-
                     let state = match key.action() {
                         KeyAction::Down => event::ElementState::Pressed,
                         KeyAction::Up => event::ElementState::Released,
                         _ => event::ElementState::Released,
                     };
-                    #[allow(deprecated)]
+
+                    #[cfg(feature = "android-native-activity")]
+                    let (keycode_u32, scancode_u32) = unsafe {
+                        // We abuse the fact that `android_activity`'s `KeyEvent` is `repr(transparent)`
+                        let event = (key as *const android_activity::input::KeyEvent<'_>).cast::<ndk::event::KeyEvent>();
+                        // We use the unsafe function directly because we want to forward the
+                        // keycode value even if it doesn't have a variant defined in the ndk
+                        // crate.
+                        (
+                            AKeyEvent_getKeyCode((*event).ptr().as_ptr()) as u32,
+                            (*event).scan_code() as u32
+                        )
+                    };
+                    #[cfg(feature = "android-game-activity")]
+                    let (keycode_u32, scancode_u32) = (key.keyCode as u32, key.scanCode as u32);
+                    let keycode = keycode_u32
+                        .try_into()
+                        .unwrap_or(ndk::event::Keycode::Unknown);
```

Comment:
I think it should be possible to get the keycode and scancode via `key.key_code()` and `key.scan_code()` without needing these backend-specific tricks.

It wasn't really intentional that the public API lets you deref the key into a `GameActivityKeyEvent` to be able to access `key.keyCode` like that - that's a mistake.